superset/assets/javascripts/explore/components/ExploreViewContainer.jsx

```diff
@@ -101,13 +106,43 @@ class ExploreViewContainer extends React.Component {
     }
   }
 
+  addHistory(isReplace = false, title) {
+    const { payload } = createExplore(this.props.form_data);
+    const longUrl = getExploreLongUrl(this.props.form_data);
+    if (isReplace) {
+      history.replaceState(
+        payload,
+        document.title,
+        longUrl);
+    } else {
+      history.pushState(
+        payload,
+        document.title,
+        longUrl);
+    }
+
+    // it seems browsers don't support pushState title attribute
+    if (title) {
+      document.title = title;
+    }
+  }
+
   handleResize() {
     clearTimeout(this.resizeTimer);
     this.resizeTimer = setTimeout(() => {
       this.setState({ height: this.getHeight(), width: this.getWidth() });
     }, 250);
   }
 
+  handleNavigate() {
```

Comment:
`Navigate` is a little ambiguous, what do you think about mirroring the listener: `handlePopstate`?

Reply:
fixed.